todo_app/data/session_items.py

```diff
@@ -85,85 +98,108 @@ def get_items():
     return sorted_default_items
 
 def get_board_details():
-    
-    boardname='todo items'
-    if(os.environ.get('TRELLO_BOARD_ID') !=None):
+    global db
+    if(db==None):
+        db=OpenMongo()
+
+    result=db.list_collection_names()
+    if(len(result)==0):
+        boards_name = {"name": "todo items","date": datetime.utcnow()}
+        boardscol = db["boards"]
+        post_id = boardscol.insert_one(boards_name).inserted_id
         return {
-        "id": os.environ['TRELLO_BOARD_ID'],
-        "name": "To Service"
+        "id": str(ObjectId(post_id)),
+        "name": "todo items",
         }
-
-    url="https://api.trello.com/1/search"
-
-    headers = {
-    "Accept": "application/json"
-    }
-
-    query = {
-    'key': os.getenv('TRELLO_KEY'),
-    'token': os.getenv('TRELLO_TOKEN'),  
-    'query' : 'name:"'+boardname+'"',    
-    'modelTypes' : 'boards',
-    'board_fields' : '"name"'
+    else:
+        for coll in result:
+            if(coll=="boards"):
+                document=db.boards.find_one({"name": "todo items"})
+                return {
+                "id": str(ObjectId(document['_id'])),
+                "name": document["name"],
+                }
+            
+    boards = {"name": "todo items","date": datetime.utcnow()}
+    boardscol = db["boards"]
+    post_id = boardscol.insert_one(boards).inserted_id
+    return {
+    "id": str(ObjectId(post_id)),
+    "name": "todo items",
     }
 
-    r = requests.request(
-    "GET",
-    url,
-    headers=headers,
-    params=query
-    )
-    #print(query)
-    if r.status_code == 200:
-        jsonResponse = r.json()
-        return {
-        "id": jsonResponse["boards"][0]["id"],
-        "name": boardname,
-        }
-    else:        
+    if(post_id==None):        
         raise CustomError('Error finding a Trello board')
    
-
+    
 def get_listinboard(board_id):
+    global db
+    if(db==None):
+        db=OpenMongo()
+
     board_lists = []   
-     
-    r = requests.get('https://api.trello.com/1/boards/'+board_id+'/lists', params={'key': os.getenv('TRELLO_KEY'), 'token': os.getenv('TRELLO_TOKEN')})
-    if r.status_code == 200:
-        jsonResponse = r.json()  
-        for json_item in jsonResponse:
+    result=db.list_collection_names()
+    if(len(result)==0):
```

Comment:
Could do `db['lists'].count_documents({}) == 0`, if the collection doesn't exist then it's considered empty.

This all works, but is pretty complicated. Up to you, but you can actually get away without having a concept of lists at all and just having a status on the card objects again.